### PR TITLE
Match statement replacement in stdlib modules

### PR DESCRIPTION
**Pull request by ThisaruGuruge**

### CODE REVIEW

File: stdlib/privacy/src/main/ballerina/privacy/database_pii_store_utils.bal [hunk 1]
```diff
@@ -77,18 +77,15 @@ function validateFieldName (string tableName, string idColumn, string piiColumn)
...
-        }
-        error err => {
+    if (queryResult is int) {
+        if (queryResult > 0) {
+            return id;
+        } else {
+            error err = error("Unable to insert PII with identifier " + id);
             return err;
         }
+    } else {
```
Comment (MaryamZi) [inline stdlib/privacy/src/main/ballerina/privacy/database_pii_store_utils.bal]: Can't we use check at the beginning and return the error there instead, with such functions?

```bal
         int intResult = check queryResult;
         if (intResult > 0) {
             return id;
         } else {
             error err = error("Unable to insert PII with identifier " + id);
             return err;
         }
```

File: stdlib/bir/src/main/ballerina/bir/bir_func_body_parser.bal [hunk 2]
```diff
@@ -135,12 +135,11 @@ public type FuncBodyParser object {
...
-        VariableDcl dcl => return dcl;
-        () => {
-            error err = error("local var missing " + varName);
-            panic err;
-        }
+    if (posibalDcl is VariableDcl) {
+        return posibalDcl;
+    } else {
+        error err = error("local var missing " + varName);
+        panic err;
```
[the remark this answers is not in the document]
Comment (ThisaruGuruge) [reply stdlib/bir/src/main/ballerina/bir/bir_func_body_parser.bal]: I do not have a complete idea of the scenario here.
@manuranga WDYT ?

[the remark this answers is not in the document]
Comment (manuranga) [reply stdlib/bir/src/main/ballerina/bir/bir_func_body_parser.bal]: This panic should never occur outside development. It's there as an 'assert' statement. So panic should be fine.

File: stdlib/io/src/main/ballerina/io/Module.md [hunk 3]
```diff
@@ -102,13 +76,10 @@ var readableRecordsChannel = new io:ReadableTextRecordChannel(readableCharChanne
...
-        string[] record => {
-            println(record); // Retrieved a record.
-        }
-        error err => {
-            return err; // An IO error occurred when reading the records.
-        }
+    if (result is string[]) {
+        println(record); // Retrieved a record.
+    } else {
+        return err; // An IO error occurred when reading the records.
```
Comment (Anoukh) [inline stdlib/io/src/main/ballerina/io/Module.md]: should this be result?
```suggestion
        return result; // An IO error occurred when reading the records.
```

File: stdlib/grpc/src/main/ballerina/grpc/Module.md [hunk 4]
```diff
@@ -129,14 +126,11 @@ public function main (string... args) {
...
-         // If the service returns an error, print the error.
-        error payloadError => {
-            io:println("Error occured while sending event " + payloadError.message);
-        }
-        () => {
-            io:println("Connected successfully to service");
-        }
+    if (result is error) {
+        // If the service returns an error, print the error.
+        io:println("Error occured while sending event " + payloadError.message);
```
Comment (Anoukh) [inline stdlib/grpc/src/main/ballerina/grpc/Module.md]: Should this be result.message ?
```suggestion
        io:println("Error occured while sending event " + result.message);
```

File: stdlib/grpc/src/main/ballerina/grpc/Module.md [hunk 5]
```diff
@@ -74,18 +74,15 @@ headers.setEntry("id", "newrequest1");
...
-   // If an error is returned, print the error message.
-   error err => {
-       io:println("Error while connecting grpc end-point : " + err.message);
-   }
+if (responseFromServer is (string, grpc:Headers)) {
+    // If a response is received, print the payload.
+    string result;
+    grpc:Headers resHeaders;
+    (result, resHeaders) = responseFromServer;
+    io:println("Response received : " + result);
```
Comment (Anoukh) [inline stdlib/grpc/src/main/ballerina/grpc/Module.md]: I think we can simply use index based access here
```suggestion
    io:println("Response received : " + responseFromServer[0]);
```

File: stdlib/grpc/src/main/ballerina/grpc/Module.md [hunk 6]
```diff
@@ -74,18 +74,15 @@ headers.setEntry("id", "newrequest1");
...
+} else {
+    // If an error is returned, print the error message.
+    io:println("Error while connecting grpc end-point : " + err.message);
```
Comment (Anoukh) [inline stdlib/grpc/src/main/ballerina/grpc/Module.md]:
```suggestion
    io:println("Error while connecting grpc end-point : " + responseFromServer.message);
```

**Pull request merged**